File: CS/CustomProgressIndicator/MyProgressIndicationForm.cs
using System.Windows.Forms;
using DevExpress.XtraEditors;
// ...

namespace CustomProgressIndicator {
    public partial class MyProgressIndicationForm : XtraForm {
        public MyProgressIndicationForm() {
            InitializeComponent();
        }

        int minValue;
        int maxValue;

        private new DevExpress.XtraGauges.Win.Gauges.Linear.LinearScaleComponent Scale { get { return this.linearGauge1.Scales[0]; } }
        private DevExpress.XtraGauges.Win.Base.LabelComponent Label { get { return linearGauge1.Labels[0]; } }

        public void Begin(int maxValue) { Begin(0, maxValue, 0); }
        public void Begin(int minValue, int maxValue, int currentValue) {
            this.minValue = minValue;
            this.maxValue = maxValue;
            SetProgress(currentValue);
        }

        public void SetProgress(int currentValue) {
            float percent =
                100f * (currentValue - this.minValue) / (this.maxValue - this.minValue);
            Scale.Value = percent;
            Label.Text = string.Format("{0}", (int)percent);
        }
    }
}

File: VB/CustomProgressIndicator/MyProgressIndicationForm.vb
Imports DevExpress.XtraEditors

' ...
Namespace CustomProgressIndicator

    Public Partial Class MyProgressIndicationForm
        Inherits XtraForm

        Public Sub New()
            InitializeComponent()
        End Sub

        Private minValue As Integer

        Private maxValue As Integer

        Private Overloads ReadOnly Property Scale As DevExpress.XtraGauges.Win.Gauges.Linear.LinearScaleComponent
            Get
                Return linearGauge1.Scales(0)
            End Get
        End Property

        Private ReadOnly Property Label As DevExpress.XtraGauges.Win.Base.LabelComponent
            Get
                Return linearGauge1.Labels(0)
            End Get
        End Property

        Public Sub Begin(ByVal maxValue As Integer)
            Begin(0, maxValue, 0)
        End Sub

        Public Sub Begin(ByVal minValue As Integer, ByVal maxValue As Integer, ByVal currentValue As Integer)
            Me.minValue = minValue
            Me.maxValue = maxValue
            SetProgress(currentValue)
        End Sub

        Public Sub SetProgress(ByVal currentValue As Integer)
            Dim percent As Single = 100F * (currentValue - minValue) / (maxValue - minValue)
            Scale.Value = percent
            Label.Text = String.Format("{0}", CInt(percent))
        End Sub
    End Class
End Namespace
